File: JavaScript/boj/Silver/15652.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const [N, M] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const numbers = [];
const result = [];

const dfs = (depth, start) => {
  if (depth === M) {
    result.push(numbers.join(' '));
    return;
  }

  for (let i = start; i <= N; i++) {
    numbers.push(i);
    dfs(depth + 1, i);
    numbers.pop();
  }
};

dfs(0, 1);
console.log(result.join('\n'));

File: Python/boj/Silver/15652.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N, M = map(int, input().split(" "))
    numbers = []
    result = []

    def dfs(depth, start):
        if depth == M:
            result.append(" ".join(numbers))
            return

        for i in range(start, N + 1):
            numbers.append(str(i))
            dfs(depth + 1, i)
            numbers.pop()

    dfs(0, 1)
    print("\n".join(result))
